pymc3/distributions/posterior_predictive.py

```diff
@@ -69,15 +57,15 @@ class _TraceDict(UserDict):
     ~~~~~~~~~~
         varnames: list of strings"""
 
-    varnames: List[str]
+    varnames: list[str]
     _len: int
     data: Point
 
     def __init__(
         self,
-        point_list: Optional[List[Point]] = None,
-        multi_trace: Optional[MultiTrace] = None,
-        dict_: Optional[Point] = None,
+        point_list: list[Point] | None = None,
+        multi_trace: MultiTrace | None = None,
+        dict_: Point | None = None,
```

Comment:
This syntactic sugar is new to me - since which Python version is this supported?

Reply:
PyMC3 supports Python >=3.7... The CI was fine, but please find out exactly then - we don't want to break something by accident.

Reply:
Okay, I'll check this now and let you know.

Reply:
Hey @michaelosthege I just ran the test for posterior_predicitive  on my version of pymc3, using python 3.7, and everything works fine.

Reply:
It works fine in Python3+ as long as there's `from __future__ import annotations` at the top (and `pyupgrade` will only use this syntax in files where `from __future__ import annotations` is present)

Reply:
In 3.10 it'll become the default and'll work without `from __future__ import annotations`

Reply:
We're good to go then. Thanks!